reconcile/vault_replication.py

```diff
@@ -0,0 +1,225 @@
+import logging
+import re
+
+from reconcile.utils.vault import VaultClient, _VaultClient, SecretNotFound
+from reconcile.gql_definitions.jenkins_configs import jenkins_configs
+from reconcile.gql_definitions.jenkins_configs.jenkins_configs import (
+    JenkinsConfigV1,
+)
+
+from reconcile.gql_definitions.vault_policies import vault_policies
+from reconcile.gql_definitions.vault_instances import vault_instances
+from reconcile.gql_definitions.vault_instances.vault_instances import (
+    VaultReplicationV1,
+    VaultInstanceV1,
+)
+from reconcile.gql_definitions.vault_policies.vault_policies import (
+    VaultPolicyV1,
+)
+
+from reconcile.utils import gql
+
+from typing import List, cast, Optional
+
+QONTRACT_INTEGRATION = "vault-replication"
+
+
+def deep_copy_versions(
+    dry_run: bool,
+    source_vault,
+    dest_vault,
+    current_dest_version,
```

Comment:
Since version has a meaning in terms of Vault secrets, it would be more exact that this is `None` in the case of non-existing dest secret instead of `0`.

Reply:
Changed to None

Reply:
hmm, you're still calling it using `0` as `current_dest_version` when it is missing.

Reply:
Changed the logic to not call `deep_copy_versions` when the secret is a v1, now it's a simple write